### PR TITLE
Add support for unzipped IntelliJs in resolvers.intellij.repositories config

### PR DESCRIPTION
**Pull request by wiacekm**
Automatically recognise if it is a zip archive or directory.

Closes: #158

### CODE REVIEW

**Comment by PawelLipski** [comment]
Gr8 🎉 pls refrain from merging for a second, I'll check this version against https://github.com/VirtusLab/git-machete-intellij-plugin

**Comment by PawelLipski** [comment]
Actually, I'd suggest setting up a requirement for CI checks to pass on a commit up-to-date with PR base branch... under `Protected branches` in repo settings

**Comment by PawelLipski** [comment]
Huh looks that IDEs downloaded by intellj-plugin-verifier simply don't contain this `dependencies.txt` file:

```
$ ls -1 ~/.pluginVerifier/ides/IC-2022.1/
bin
build.txt
Install-Linux-tar.txt
jbr
lib
license
LICENSE.txt
markerFile
NOTICE.txt
plugins
product-info.json
```

But! maybe you could check `.../jbr/releease` file as a fallback?

```
 cat ~/.pluginVerifier/ides/IC-2022.1/jbr/release 
...
IMPLEMENTOR_VERSION="JBR-11.0.14.1.1-2043.25-jcef"
...
```

Parsing would need to change a bit, since `dependencies.txt` has a different format:

```
....
jdkBuild=11_0_14_1b2043.25
....
```

File: core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/IntelliJProvider.scala [hunk 1]
```diff
@@ -140,10 +144,13 @@ final case class IntelliJFactory(
   }
 
   private def installIntelliJ(version: IntelliJVersion, root: Path): Unit = {
-    println(s"Installing $version")
```
Comment (lukaszwawrzyk) [inline core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/IntelliJProvider.scala:143]: I'd like to keep this message. When probe starts it reports the progress this way (installing IJ, later installing plugins).

Comment (wiacekm) [reply core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/IntelliJProvider.scala:143]: removed

**Comment by PawelLipski** [comment]
Also! not sure if caused by this PR, rather probably by recent dependency updates:

```
   [intellij-stdout] Exception in thread "ideprobe-worker-1" 
    [intellij-stdout] java.lang.NoSuchMethodError: 'shapeless.Generic shapeless.Generic$.instance(scala.Function1, scala.Function1)'
    [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$anon$exportedReader$macro$26$1.inst$macro$1$lzycompute(Endpoints.scala:14)
    [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$anon$exportedReader$macro$26$1.inst$macro$1(Endpoints.scala:14)
    [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$.<init>(Endpoints.scala:14)
    [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$.<clinit>(Endpoints.scala)
    [intellij-stdout]   at org.virtuslab.ideprobe.BaseProbeHandlerContributor.registerHandlers(BaseProbeHandlerContributor.scala:14)
    [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.$anonfun$collectHandlers$1(ProbeHandlers.scala:38)
    [intellij-stdout]   at scala.collection.IndexedSeqOptimized.foldLeft(IndexedSeqOptimized.scala:60)
    [intellij-stdout]   at scala.collection.IndexedSeqOptimized.foldLeft$(IndexedSeqOptimized.scala:68)
    [intellij-stdout]   at scala.collection.mutable.ArrayOps$ofRef.foldLeft(ArrayOps.scala:198)
    [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.org$virtuslab$ideprobe$ProbeHandlers$$collectHandlers(ProbeHandlers.scala:38)
    [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.<init>(ProbeHandlers.scala:20)
    [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.<clinit>(ProbeHandlers.scala)
    [intellij-stdout]   at org.virtuslab.ideprobe.Probe.handler(Probe.scala:13)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.$anonfun$listen$1(JsonRpcEndpoint.scala:50)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.$anonfun$listen$1$adapted(JsonRpcEndpoint.scala:49)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcConnection.$anonfun$onRequest$1(JsonRpcConnection.scala:48)
    [intellij-stdout]   at scala.collection.Iterator.foreach(Iterator.scala:941)
    [intellij-stdout]   at scala.collection.Iterator.foreach$(Iterator.scala:941)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpc$$anon$1.foreach(JsonRpc.scala:18)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcConnection.onRequest(JsonRpcConnection.scala:46)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.listen(JsonRpcEndpoint.scala:49)
    [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.listen$(JsonRpcEndpoint.scala:48)
    [intellij-stdout]   at org.virtuslab.ideprobe.Probe.listen$lzycompute(Probe.scala:12)
    [intellij-stdout]   at org.virtuslab.ideprobe.Probe.listen(Probe.scala:12)
    [intellij-stdout]   at org.virtuslab.ideprobe.Probe$.$anonfun$start$1(Probe.scala:21)
    [intellij-stdout]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
    [intellij-stdout]   at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
    [intellij-stdout]   at scala.util.Success.$anonfun$map$1(Try.scala:255)
    [intellij-stdout]   at scala.util.Success.map(Try.scala:213)
    [intellij-stdout]   at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
    [intellij-stdout]   at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
    [intellij-stdout]   at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
    [intellij-stdout]   at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
    [intellij-stdout]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    [intellij-stdout]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    [intellij-stdout]   at java.base/java.lang.Thread.run(Thread.java:829)
    [intellij-stdout] opened Welcome to IntelliJ IDEA
    [intellij-stdout] closed com.intellij.ui.Splash
```

Repro:

```
./gradlew -Pagainst=2022.1 uiTest
```

on the current state of https://github.com/VirtusLab/git-machete-intellij-plugin/pull/834

(assuming 2022.1 is downloaded by `./gradlew runPluginVerifier` and ide-probe `0.24.0+18-b852664b-SNAPSHOT` is published locally)

**Comment by lukaszwawrzyk** [comment]
> Also! not sure if caused by this PR, rather probably by recent dependency updates:
> 
> ```
>    [intellij-stdout] Exception in thread "ideprobe-worker-1" 
>     [intellij-stdout] java.lang.NoSuchMethodError: 'shapeless.Generic shapeless.Generic$.instance(scala.Function1, scala.Function1)'
>     [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$anon$exportedReader$macro$26$1.inst$macro$1$lzycompute(Endpoints.scala:14)
>     [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$anon$exportedReader$macro$26$1.inst$macro$1(Endpoints.scala:14)
>     [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$.<init>(Endpoints.scala:14)
>     [intellij-stdout]   at org.virtuslab.ideprobe.protocol.Endpoints$.<clinit>(Endpoints.scala)
>     [intellij-stdout]   at org.virtuslab.ideprobe.BaseProbeHandlerContributor.registerHandlers(BaseProbeHandlerContributor.scala:14)
>     [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.$anonfun$collectHandlers$1(ProbeHandlers.scala:38)
>     [intellij-stdout]   at scala.collection.IndexedSeqOptimized.foldLeft(IndexedSeqOptimized.scala:60)
>     [intellij-stdout]   at scala.collection.IndexedSeqOptimized.foldLeft$(IndexedSeqOptimized.scala:68)
>     [intellij-stdout]   at scala.collection.mutable.ArrayOps$ofRef.foldLeft(ArrayOps.scala:198)
>     [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.org$virtuslab$ideprobe$ProbeHandlers$$collectHandlers(ProbeHandlers.scala:38)
>     [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.<init>(ProbeHandlers.scala:20)
>     [intellij-stdout]   at org.virtuslab.ideprobe.ProbeHandlers$.<clinit>(ProbeHandlers.scala)
>     [intellij-stdout]   at org.virtuslab.ideprobe.Probe.handler(Probe.scala:13)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.$anonfun$listen$1(JsonRpcEndpoint.scala:50)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.$anonfun$listen$1$adapted(JsonRpcEndpoint.scala:49)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcConnection.$anonfun$onRequest$1(JsonRpcConnection.scala:48)
>     [intellij-stdout]   at scala.collection.Iterator.foreach(Iterator.scala:941)
>     [intellij-stdout]   at scala.collection.Iterator.foreach$(Iterator.scala:941)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpc$$anon$1.foreach(JsonRpc.scala:18)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcConnection.onRequest(JsonRpcConnection.scala:46)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.listen(JsonRpcEndpoint.scala:49)
>     [intellij-stdout]   at org.virtuslab.ideprobe.jsonrpc.JsonRpcEndpoint.listen$(JsonRpcEndpoint.scala:48)
>     [intellij-stdout]   at org.virtuslab.ideprobe.Probe.listen$lzycompute(Probe.scala:12)
>     [intellij-stdout]   at org.virtuslab.ideprobe.Probe.listen(Probe.scala:12)
>     [intellij-stdout]   at org.virtuslab.ideprobe.Probe$.$anonfun$start$1(Probe.scala:21)
>     [intellij-stdout]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
>     [intellij-stdout]   at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
>     [intellij-stdout]   at scala.util.Success.$anonfun$map$1(Try.scala:255)
>     [intellij-stdout]   at scala.util.Success.map(Try.scala:213)
>     [intellij-stdout]   at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
>     [intellij-stdout]   at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
>     [intellij-stdout]   at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
>     [intellij-stdout]   at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
>     [intellij-stdout]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
>     [intellij-stdout]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
>     [intellij-stdout]   at java.base/java.lang.Thread.run(Thread.java:829)
>     [intellij-stdout] opened Welcome to IntelliJ IDEA
>     [intellij-stdout] closed com.intellij.ui.Splash
> ```
> 
> Repro:
> 
> ```
> ./gradlew -Pagainst=2022.1 uiTest
> ```
> 
> on the current state of [VirtusLab/git-machete-intellij-plugin#834](https://github.com/VirtusLab/git-machete-intellij-plugin/pull/834)
> 
> (assuming 2022.1 is downloaded by `./gradlew runPluginVerifier` and ide-probe `0.24.0+18-b852664b-SNAPSHOT` is published locally)

@tpasternak can you take a look?

**Comment by PawelLipski** [comment]
> Also! not sure if caused by this PR, rather probably by recent dependency updates:

Okay, I managed to circumvent that... but only by downgrading/reverting certain deps:

```scala
  val pureConfig = {
    val typesafeConfig = "com.typesafe" % "config" % "1.4.2"
    val shapeless = "com.chuusai" %% "shapeless" % "2.3.3"
    val pureConfigModules =
      Seq("pureconfig", "pureconfig-macros", "pureconfig-generic", "pureconfig-generic-base", "pureconfig-core")
    pureConfigModules.map { module =>
      "com.github.pureconfig" %% module % "0.14.1"
    } ++ Seq(typesafeConfig, shapeless)
  }
```

TBH I'd suggest just using the above library versions in `project/Dependencies.scala` to make sure the next shipped version is working... and research the problems with binary compatibility separately 🤔

**Pull request merged**